release-operator-code.sh

```diff
@@ -30,21 +30,27 @@ cd "${BASE_DIR}"
 echo
 echo "## Creating release '${RELEASE}' of the Che operator docker image"
 
+DefaultPluginRegistryImage
+
 lastDefaultCheVersion=$(grep 'DefaultCheServerImageTag' "pkg/deploy/defaults.go" | sed -e 's/.*DefaultCheServerImageTag *= *"\([^"]*\)"/\1/')
 lastDefaultKeycloakVersion=$(grep 'DefaultKeycloakUpstreamImage' "pkg/deploy/defaults.go" | sed -e 's/.*DefaultKeycloakUpstreamImage *= *"[^":]*:\([^"]*\)"/\1/')
+lastDefaultPluginRegistryVersion=$(grep 'DefaultPluginRegistryImage' "pkg/deploy/defaults.go" | sed -e 's/.*DefaultPluginRegistryImage *= *"[^":]*:\([^"]*\)"/\1/')
+lastDefaultDevfileRegistryVersion=$(grep 'DefaultDevfileRegistryImage' "pkg/deploy/defaults.go" | sed -e 's/.*DefaultDevfileRegistryImage *= *"[^":]*:\([^"]*\)"/\1/')
 if [ "${lastDefaultCheVersion}" != "${lastDefaultKeycloakVersion}" ]
 then
   echo "#### ERROR ####"
   echo "Current default Che version: ${lastDefaultCheVersion}"
   echo "Current default Keycloak version: ${lastDefaultKeycloakVersion}"
-  echo "Current default version for Che and keycloak are not the same in file 'pkg/deploy/defaults.go'."
+  echo "Current default Devfile Registry version: ${lastDefaultDevfileRegistryVersion}"
+  echo "Current default Plugin Registry version: ${lastDefaultPluginRegistryVersion}"
+  echo "Current default version for various Che containers are not the same in file 'pkg/deploy/defaults.go'."
```

Comment:
why can't the default versions differ? what if ther's a 7.0.1 version of the plugin registry, and 7.0.3 version of the devfile registry, and che/keycloak are still on 7.0.0 ?

Or what happens in the CRW case there the version of SSO is 7.3.3 but the version of Che/CRW server is 2.0?

Reply:
> why can't the default versions differ? what if ther's a 7.0.1 version of the plugin registry, a 7.0.3 version of the devfile registry, and che/keycloak are still on 7.0.0 ?

Afaik, as concerning **upstream Che**, we expect the releases of `che-server`,  `che-keycloak`, `che-plugin-registry` and `che-devfile-registries` to be synchronized. That's the whole point of the new Che release process, isn't it @l0rd ?

> Or what happens in the CRW case there the version of SSO is 7.3.3 but the version of Che/CRW server is 2.0? I expect that in CRW we may release registry container updates more frequently (eg., weekly? once per sprint?) while CRW updates may be slower (every 6 weeks? every 12 weeks?) so these number will likely diverge.

This script is tor preparing a new release of **upstream Che** operator.
In the context of CRW 2.0, we might need to distinguish defaults between `che` and `codeready` flavours for registries as well (the same as we already did for other docker images). Then CRW releasing script would perform replacements in this file as it is convenient for the CRW-specific release process.

But I'd prefer letting this for a distinct PR.

Reply:
Agreed, I was just making sure we're not boxed into a corner or forced to fork downstream. :)

Reply:
Great. Thanks.

Reply:
Yes for simplicity the versions of the servers (che, postgres, keycloak, registries) should always been in sync when releasing. 

If for instance we release a new version of che-server we will release a new version of che-devfile-registry even if there hasn't been any change on the registry side.